shaders_js.go

```diff
@@ -0,0 +1,83 @@
+//+build netgo
+
+package engo
+
+import (
+	"github.com/gopherjs/gopherjs/js"
+)
+
+type Shader interface {
+	Initialize(width, height float32)
+	Pre()
+	Draw(texture *js.Object, buffer *js.Object, x, y, rotation float32)
```

Comment:
This would be where our `EngoEngine/gl` package would come in handy. We'd use `gl.Texture` here, and what it is beneath the hood isn't relevant to us. For the desktop, it's a `uint32`. For the web, it's `*js.Object`. 

I feel like it'd be nicer to use that, so we won't have to split the shaders in two. (because I don't see any other differences? )